File: TeeNew/Font_Gradient.pas
unit Font_Gradient;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeProcs, TeEngine, Chart, Series;

type
  TFontGradient = class(TBaseForm)
    Button1: TButton;
    CheckBox1: TCheckBox;
    CheckBox2: TCheckBox;
    procedure Button1Click(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure CheckBox2Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeEdiGrad, TeCanvas;

procedure TFontGradient.Button1Click(Sender: TObject);
begin
  EditTeeGradient(Self,Chart1.Title.Font.Gradient);
end;

procedure TFontGradient.CheckBox1Click(Sender: TObject);
begin
  Chart1.Title.Font.Gradient.Visible:=CheckBox1.Checked;

  CheckBox2.Enabled:=CheckBox1.Checked;
end;

procedure TFontGradient.FormCreate(Sender: TObject);
begin
  inherited;

  // Activate Font gradient:

  with Chart1.Title.Font.Gradient do
  begin
    Visible:=True;
    StartColor:=clNavy;
    MidColor:=clRed;
    EndColor:=clYellow;
    Direction:=gdLeftRight; // Note: Requires using "TeCanvas.pas" unit
  end;
end;

procedure TFontGradient.CheckBox2Click(Sender: TObject);
begin
  Chart1.Title.Font.Outline.Fill.Gradient.Visible:=CheckBox2.Checked;
end;

initialization
  RegisterClass(TFontGradient);
end.

File: TeeNew/Builder2007/Font_Gradient.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Font_Gradient.h"

#include "TeeEdiGrad.hpp"
#include "TeCanvas.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TFontGradient *FontGradient;
//---------------------------------------------------------------------------
__fastcall TFontGradient::TFontGradient(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TFontGradient::Button1Click(TObject *Sender)
{
  EditTeeGradient(this,Chart1->Title->Font->Gradient,false);
}
//---------------------------------------------------------------------------

void __fastcall TFontGradient::CheckBox1Click(TObject *Sender)
{
  Chart1->Title->Font->Gradient->Visible = CheckBox1->Checked;
  CheckBox2->Enabled = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TFontGradient::FormCreate(TObject *Sender)
{
  // Activate Font gradient:

  TTeeGradient *grad = Chart1->Title->Font->Gradient;
  grad->Visible = true;
  grad->StartColor = clNavy;
  grad->MidColor = clRed;
  grad->EndColor = clYellow;
  grad->Direction = gdLeftRight; // Note: Requires using "TeCanvas.pas" unit
}
//---------------------------------------------------------------------------

void __fastcall TFontGradient::CheckBox2Click(TObject *Sender)
{
  Chart1->Title->Font->Gradient->Outline = CheckBox2->Checked;
}
//---------------------------------------------------------------------------

